dev/src/DocFx/Command/DocFx.php

```diff
@@ -0,0 +1,291 @@
+    {
+        $component = $input->getOption('component') ?: basename(getcwd());
+        $componentPath = $this->checkComponent($component);
+        $output->writeln(sprintf('Generating documentation for <options=bold;fg=white>%s</>', $component));
+        $xml = $input->getOption('xml');
+        $outDir = $input->getOption('out');
+        if (empty($xml)) {
+            $output->write('Running phpdoc to generate structure.xml... ');
+            // Run "phpdoc"
+            $process = new Process([
+                'phpdoc',
+                '-d',
+                sprintf('%s/src', $componentPath),
+                '--template',
+                'xml',
+                '--target',
+                $outDir
+            ]);
+            $process->mustRun();
+            $output->writeln('Done.');
+            $xml = $outDir . '/structure.xml';
+        }
+        if (!file_exists($xml)) {
+            throw new \Exception($input->getOption('xml') ? 'Unable to load provided structure.xml'
+                : sprintf('Default structure.xml file "%s" not found.', $xml));
+        }
+
+        if (!file_exists($xml)) {
+            throw new RuntimeException('provided path to structure.xml does not exist');
+        }
+
+        $releaseLevel = $this->getReleaseLevel();
+        $namespaces = $this->getNamespaces();
+
+        if (!is_dir($outDir)) {
+            if (!mkdir($outDir)) {
+                throw new RuntimeException('out directory doesn\'t exist and cannot be created');
+            }
+        }
+
+        $output->write(sprintf('Writing output to <fg=white>%s</>... ', $outDir));
+
+        // YAML dump configuration
+        $inline = 9; // The level where you switch to inline YAML
+        $indent = 2; // The amount of spaces to use for indentation of nested nodes
+        $flags = Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK;
+
+        $tocItems = [];
+        foreach ($namespaces as $namespace) {
+            $pages = new Pages($xml, $namespace);
+            $pageList = $pages->getPages();
+            XrefTrait::$protoPackagesToPhpNamespaces = $pages->getProtoPackages();
+
+            foreach ($pageList as $page) {
+                $docFxArray = ['items' => $page->getItems()];
+
+                // Dump the YAML for the class node
+                $yaml = '### YamlMime:UniversalReference' . PHP_EOL;
+                $yaml .= Yaml::dump($docFxArray, $inline, $indent, $flags);
+
+                // Write the YAML to a file
+                $outFile = sprintf('%s/%s.yml', $outDir, $page->getFilename());
+                file_put_contents($outFile, $yaml);
+            }
+
+            $tocItems = array_merge($tocItems, $pages->getTocItems());
+        }
+
+        // Add an "overview" file if it exists
+        $overviewFile = sprintf('%s/README.md', $componentPath);
+        if ($releaseLevel === 'beta') {
+            $overviewFile = $this->getBetaNotice() . $overviewFile;
```

Comment:
How does this work, where the "beta notice" is prepended to the "overview" file path?

Reply:
I've actually commented this out because it was breaking the markdown parsing. I was hoping to add a "Beta notice" banner on the Overview page [as can be seen here](https://cloud.google.com/go/docs/reference/cloud.google.com/go/aiplatform/latest/apiv1beta1)